src/components/main/Tool.js

```diff
@@ -29,25 +30,37 @@ import "@/components/main/Tool.css";
 // object. `x` is unchanged by this operation, and `y` is a different object
 // than `x`.
 
-// Compute likely latest possible date of available data = current date - 15 d.
-// This allows for cron jobs that run in first half of month.
-// Subtract fewer/more days if cron jobs run earlier/later in month.
-// But it is not guaranteed that there is data for this date; that can only be
-// determined by consulting the backend.
-const latestPossibleDataDate = moment().subtract(15, "days");
-
 export default function Tool() {
   const config = useConfigContext();
 
   // Application state
   const [variable, setVariable] = useState(config.ui.variableSelector.initial);
   const [dataset, setDataset] = useState(config.ui.datasetSelector.initial);
-  const { isPending: dateIsPending, date, setDate } = useDateState();
+  const {
+    isPending: dateIsPending,
+    lastDateWithMonthlyData,
+    date,
+    setDate,
+  } = useDateState();
 
   // Server state
   const { isPending: baselineIsPending } = useBaseline(variable, date);
   const { isPending: monthlyIsPending } = useMonthly(variable, date);
 
+  const minYear = config.ui.yearSelector.minYear;
+  const minDate = moment({ year: minYear });
+  // There's a tricky and unresolved problem when dates are incremented beyond
+  // lastDateWithMonthlyData: the stations displayed on the map get out of sync
+  // with the actual date and data. This prevents such incrementing, and so the
+  // erroneous maps, but it does not fix the bug causing the error with the station
+  // markers.
+  const maxDate = lastDateWithMonthlyData;
+  const clipDate = (date) => clipMoment(date, minDate, maxDate);
```

Comment:
I tested incrementing the dates beyond lastDateWithMonthlyData, and I can confirm that this solution works as intended.